zotero/src/data_structure/item/mod.rs

```diff
@@ -192,6 +202,289 @@ pub struct Item {
+                    &d.date_enacted
+                }
+                ItemType::EncyclopediaArticle(d) => {
+                    &d.date
+                }
+                ItemType::Bill(d) => {
+                    &d.date
+                }
+                ItemType::Case(d) => {
+                    &d.date_decided
+                }
+                ItemType::Hearing(d) => {
+                    &d.date
+                }
+                ItemType::ConferencePaper(d) => {
+                    &d.date
+                }
+                ItemType::ForumPost(d) => {
+                    &d.date
+                }
+                ItemType::Webpage(d) => {
+                    &d.date
+                }
+                ItemType::NewspaperArticle(d) => {
+                    &d.date
+                }
+                ItemType::RadioBroadcast(d) => {
+                    &d.date
+                }
+                ItemType::InstantMessage(d) => {
+                    &d.date
+                }
+                ItemType::DictionaryEntry(d) => {
+                    &d.date
+                }
+                ItemType::Presentation(d) => {
+                    &d.date
+                }
+                ItemType::Manuscript(d) => {
+                    &d.date
+                }
+                ItemType::BlogPost(d) => {
+                    &d.date
+                }
+                ItemType::TvBroadcast(d) => {
+                    &d.date
+                }
+                ItemType::Patent(d) => {
+                    &d.issue_date
+                }
+                ItemType::Email(d) => {
+                    &d.date
+                }
+                ItemType::VideoRecording(d) => {
+                    &d.date
+                }
+                ItemType::Thesis(d) => {
+                    &d.date
+                }
+                ItemType::ComputerProgram(d) => {
+                    &d.date
+                }
+                ItemType::Podcast(d) => {
+                    &d.access_date
+                }
+                ItemType::AudioRecording(d) => {
+                    &d.date
+                }
+                ItemType::BookSection(d) => {
+                    &d.access_date
+                }
+                ItemType::Interview(d) => {
+                    &d.date
+                }
+                ItemType::Film(d) => {
+                    &d.date
+                }
+                ItemType::Attachment(d) => {
+                    &d.date_added
+                }
+                ItemType::Note(d) => {
+                    &d.date_added
+                }
+        };
+        convert_zotero_date_str(&date_str)
+    }
+}
+
+fn convert_zotero_date_str(date_str: &str) -> DateTime<Local> {
+    let date_regex = Regex::new(r"(\d{4})-(\d{2})-(\d{2})").unwrap();
+    let date_captures = date_regex.captures(date_str);
+
+    let formatter_regex = Regex::new(r"(\d{2})/(\d{4})").unwrap();
```

Comment:
We should move this regex inside a lazy_static to avoid building the regex each time the function is called.

https://stackoverflow.com/questions/35169259/how-to-make-a-compiled-regexp-a-global-variable